### PR TITLE
EZP-27908: As an Editor, I want to edit a Selection field type with the new UI guidelines

### PR DESCRIPTION
**Pull request by dew326**
**Jira ticket:** https://jira.ez.no/browse/EZP-27908

**Screencast:**
select-multiple
select-single

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/fieldType/ezselection.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            container.querySelector('.ez-data-source__options').classList.add('ez-data-source__options--hidden');
+        };
+        const handleClickOnInput = (event) => {
+            if (event.target.classList.contains('remove-selection')) {
+                handleSelection(event.target.closest('li'), false);
+
+                return;
+            }
+
+            container.querySelector('.ez-data-source__options').classList.toggle('ez-data-source__options--hidden');
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/ezselection.js]: These 2 selectors could be cached.

Comment (sunpietro) [reply src/bundle/Resources/public/js/scripts/fieldType/ezselection.js]: Please double check other cases :)

File: src/bundle/Resources/public/js/scripts/fieldType/ezselection.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                invalidStateSelectors: [SELECTOR_FIELD],
+                errorNodeSelectors: ['.ez-field-edit__label-wrapper'],
+            },
+        ],
+    });
+
+    validator.init();
+
+    [...doc.querySelectorAll(SELECTOR_FIELD)].forEach(container => {
+        const createSelectedItem = (value, label) => `<li data-value="${value}">${label}<span class="remove-selection"></span></li>`;
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/ezselection.js]: Did you forget adding a CSS class name to the `li` element?

File: src/bundle/Resources/public/scss/fieldType/_ezselection.scss [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            transition: all .2s $ez-admin-transition;
+            width: 50%;
+            border: 1px solid $ez-ground-primary;
+            background-color: $ez-white;
+
+            &--hidden {
+                transform: scaleY(0);
+            }
+
+            li {
```
Comment (sunpietro) [inline src/bundle/Resources/public/scss/fieldType/_ezselection.scss]: Please, avoid so generic selectors. Preferably, there should be CSS classes used.

File: src/bundle/Resources/views/fieldtypes/ezselection.html.twig [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{%- block ezplatform_fieldtype_ezselection_widget -%}
+    {% set attr = attr|merge({'hidden': 'hidden'}) %}
+    {{ block('choice_widget') }}
+
+    <ul class="ez-data-source__selected">
+        {% for choice in choices if choice is selectedchoice(value) %}
+            <li data-value="{{ choice.value }}">{{ choice_translation_domain is same as(false) ? choice.label : choice.label|trans({}, choice_translation_domain) }}<span class="remove-selection"></span></li>
```
Comment (sunpietro) [inline src/bundle/Resources/views/fieldtypes/ezselection.html.twig]: Can this be splitted into multiple lines for better readability?

Comment (sunpietro) [reply src/bundle/Resources/views/fieldtypes/ezselection.html.twig]: Also, it might be useful to add a CSS class to the `li` element

**Comment by dew326** [comment]
@m-tyrala fixed

**Pull request merged**